player.lua

```diff
@@ -65,19 +67,29 @@ function Player:update(dt)
 	--movement
 	self.vx = self.controller:leftAnalogMove() * Player.move_vel
 
+	if self.respawnTime <= 0 then
+		self.respawn = false
+		self.respawnTime = 2
```

Comment:
We don't need to set the respawn time back to 2 because you do it when you die.

then you can just simplify to: `self.respawn = respawnTime > 0`